tests/test_misc.py

```diff
@@ -2728,3 +2728,61 @@ def test_stkman_only(revault_network, bitcoind):
         rn.activate_vault(v)
     rn.unvault_vaults_anyhow(vaults)
     rn.cancel_vault(vaults[0])
+
+
+def test_getserverstatus(revault_network, bitcoind):
+    rn = revault_network
+    rn.deploy(n_stakeholders=2, n_managers=1)
+
+    # The coordinator is alive
+    for w in rn.participants():
+        res = w.rpc.call("getserverstatus")
+        assert res["coordinator"]["reachable"]
+        assert res["coordinator"]["host"] == f"127.0.0.1:{rn.coordinator_port}"
+
+    # The cosigners are alive, but only the managers see them
+    for w in rn.mans():
+        res = w.rpc.call("getserverstatus")
+        for cosigner in res["cosigners"]:
+            assert cosigner["reachable"]
+            # Sadly we don't persist the cosigner ports
+            assert cosigner["host"].startswith("127.0.0.1:")
+
+    # Stakeholders don't have cosigners info
+    for w in rn.stks():
+        res = w.rpc.call("getserverstatus")
+        assert res["cosigners"] == []
+
+    # Only stakeholders have info about the watchtowers...
+    for w in rn.stks():
+        res = w.rpc.call("getserverstatus")
+        # ...And well, they're dead
+        for watchtower in res["watchtowers"]:
+            assert not watchtower["reachable"]
+
+    # Managers don't have watchtowers info
+    for w in rn.mans():
+        res = w.rpc.call("getserverstatus")
+        assert res["watchtowers"] == []
+
+    # Alright, let's kill everything :D
+
+    # Hacky... We actually don't want to kill the wallets,
+    # so we kill the first N + 1 servers, which happen to be the
+    # coordinator and the N cosigners
+    for server in rn.daemons[: 1 + len(rn.stks())]:
```

Comment:
Was going to say it's brittle but i don't mind hacks in tests. But it's actually wrong! https://github.com/revault/revaultd/blob/8098371d13124b51f5c340932a4b1c52c43eb5e1/tests/test_framework/revault_network.py#L220-L262

Reply:
It's not :) 
The first daemon is the coordinatord: https://github.com/revault/revaultd/blob/8098371d13124b51f5c340932a4b1c52c43eb5e1/tests/test_framework/revault_network.py#L197
Then, N cosigners: https://github.com/revault/revaultd/blob/8098371d13124b51f5c340932a4b1c52c43eb5e1/tests/test_framework/revault_network.py#L262, https://github.com/revault/revaultd/blob/8098371d13124b51f5c340932a4b1c52c43eb5e1/tests/test_framework/revault_network.py#L311
Then, the wallets: https://github.com/revault/revaultd/blob/8098371d13124b51f5c340932a4b1c52c43eb5e1/tests/test_framework/revault_network.py#L336

Reply:
Shit, i'm definitely not woken up today.